src/test/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelEventGeneratorTest.java

```diff
@@ -0,0 +1,219 @@
+
+        when(supplierChangeAttribute.get()).thenReturn(changeAttribute);
+        when(supplierPatchSetAttribute.get()).thenReturn(patchSetAttribute);
+    }
+
+    private void setUpHostNameExceptionMock() {
+        mockStatic(InetAddress.class);
+        try {
+            when(InetAddress.getLocalHost()).thenThrow(UnknownHostException.class);
+        } catch (Exception ignore) {
+        }
+    }
+
+    private void setUpRepoUriExceptionMocks() {
+        try {
+            whenNew(URI.class).withArguments(URL).thenThrow(URISyntaxException.class);
+        } catch (Exception ignore) {
+        }
+    }
+
+    private void setUpSshBaseUrlExceptionMocks() {
+        try {
+            URI uri = new URI(URL);
+            String host = uri.getHost();
+            whenNew(URI.class).withArguments(URL).thenReturn(uri);
+            whenNew(URI.class).withArguments("ssh", null, host, DEFAULT_PORT, "/", null, null)
+                              .thenThrow(URISyntaxException.class);
+        } catch (Exception ignore) {
+        }
+    }
+
+    private void populateChangeMergedEvent() {
+        changeMergedEvent.newRev = "00000000-0000-0000-0000-000000000000";
+        changeMergedEvent.change = supplierChangeAttribute;
+        changeAttribute.project = "my-project";
+        changeAttribute.branch = "my-branch";
+        changeAttribute.url = "http://my-url.com";
+        changeMergedEvent.patchSet = supplierPatchSetAttribute;
+        patchSetAttribute.author = accountAttribute;
+        accountAttribute.username = "my-user";
+        accountAttribute.email = "my@email.com";
+    }
+
+    private void populatePatchSetCreatedEvent() {
+        patchSetCreatedEvent.change = supplierChangeAttribute;
+        changeAttribute.project = "my-project";
+        changeAttribute.branch = "my-branch";
+        changeAttribute.url = "http://my-url.com";
+        patchSetCreatedEvent.patchSet = supplierPatchSetAttribute;
+        patchSetAttribute.revision = "00000000-0000-0000-0000-000000000000";
+        patchSetAttribute.author = accountAttribute;
+        accountAttribute.username = "my-user";
+        accountAttribute.email = "my@email.com";
+    }
+
+    private void verifyEiffelSourceChangeSubmittedEvent(
+            EiffelSourceChangeSubmittedEvent eiffelEvent) {
+        JsonObject eiffelEventJson = gson.toJsonTree(eiffelEvent).getAsJsonObject();
+        LOGGER.debug("EiffelSourceChangeSubmittedEvent : \n{}", eiffelEventJson.toString());
+
+        JsonObject meta = eiffelEventJson.getAsJsonObject("msgParams").getAsJsonObject("meta");
+        JsonObject source = meta.getAsJsonObject("source");
+        JsonObject data = eiffelEventJson.getAsJsonObject("eventParams").getAsJsonObject("data");
+        JsonObject submitter = data.getAsJsonObject("submitter");
+        JsonObject gitIdentifier = data.getAsJsonObject("gitIdentifier");
+
+        assertTrue(meta.get("type").getAsString().equals(SCS_EVENT));
```

Comment:
If a assert fails how do you troubleshoot? Wouldn't it be better to let the assert tell us what is wrong? Maybe use something like this https://dev.to/code2bits/writing-assertions-for-string-objects-using-hamcrest-and-junit-33d0

Reply:
We can add expected and actual assert and also an error message as the first of 3 arguments.

Reply:
Done.